tests/functional/adapter/seed_column_types/fixtures.py

```diff
@@ -0,0 +1,111 @@
+_MACRO_TEST_IS_TYPE_SQL = """
+{% macro simple_type_check_column(column, check) %}
+    {% if check == 'string' %}
+        {{ return(column.is_string()) }}
+    {% elif check == 'float' %}
+        {{ return(column.is_float()) }}
+    {% elif check == 'number' %}
+        {{ return(column.is_number()) }}
+    {% elif check == 'numeric' %}
+        {{ return(column.is_numeric()) }}
+    {% elif check == 'integer' %}
+        {{ return(column.is_integer()) }}
+    {% else %}
+        {% do exceptions.raise_compiler_error('invalid type check value: ' ~ check) %}
+    {% endif %}
+{% endmacro %}
+
+{% macro type_check_column(column, type_checks) %}
+    {% set failures = [] %}
+    {% for type_check in type_checks %}
+        {% if type_check.startswith('not ') %}
+            {% if simple_type_check_column(column, type_check[4:]) %}
+                {% do log('simple_type_check_column got ', True) %}
+                {% do failures.append(type_check) %}
+            {% endif %}
+        {% else %}
+            {% if not simple_type_check_column(column, type_check) %}
+                {% do failures.append(type_check) %}
+            {% endif %}
+        {% endif %}
+    {% endfor %}
+    {% if (failures | length) > 0 %}
+        {% do log('column ' ~ column.name ~ ' had failures: ' ~ failures, info=True) %}
+    {% endif %}
+    {% do return((failures | length) == 0) %}
+{% endmacro %}
+
+{% test is_type(seed, column_map) %}
+    {% if not execute %}
+        {{ return(None) }}
+    {% endif %}
+    {% if not column_map %}
+        {% do exceptions.raise_compiler_error('test_is_type must have a column name') %}
+    {% endif %}
+    {% set columns = adapter.get_columns_in_relation(seed) %}
+    {% if (column_map | length) != (columns | length) %}
+        {% set column_map_keys = (column_map | list | string) %}
+        {% set column_names = (columns | map(attribute='name') | list | string) %}
+        {% do exceptions.raise_compiler_error('did not get all the columns/all columns not specified:\n' ~ column_map_keys ~ '\nvs\n' ~ column_names) %}
+    {% endif %}
+    {% set bad_columns = [] %}
+    {% for column in columns %}
```

Comment:
This is a long macro. I would consider breaking it into smaller macros. For instance, the body of this for loop could be defined a separate macro and referenced here. So you'd have a macro like this:
```
{% macro is_bad_column(column, column_map) %}
    {% set column_key = (column.name | lower) %}
    {% if column_key not in column_map %}
        {% do exceptions.raise_compiler_error('column key ' ~ column_key ~ ' not found in ' ~ (column_map | list | string)) %}
    {% endif %}

    {% set type_checks = column_map[column_key] %}
    {% if not type_checks %}
        {% do exceptions.raise_compiler_error('no type checks?') %}
    {% endif %}
    {{ return(not type_check_column(column, type_checks)) }}
{% endmacro %}
```
And then the for loop would look like this:
```
    ...
    {% for column in columns %}
        {% if is_bad_column(column, column_map) %}
            {% do bad_columns.append(column.name) %}
        {% endif %}
    {% endfor %}
    ...
```

Reply:
can we break up tests? haven't actually done that to my knowledge I assume they should work like regular macros hmm

Reply:
If we're concerned about breaking the code by not writing the tests in a particular way, I would argue that's a good test to write.

Reply:
You can break it up by having certain segments be `statements` or even other macros that are called by the test macro.